Content/Scripts/abn_tree_directive.js

```diff
@@ -205,17 +205,26 @@
                           }
                       });
                       add_branch_to_list = function (level, branch, visible) {
-                          var child, child_visible, tree_icon, _i, _len, _ref, _results, extra_class;
+                          var child, child_visible, tree_icon, _i, _len, _ref, _results, extra_class, resource_icon;
                           if (branch.expanded == null) {
                               branch.expanded = false;
                           }
+
+                          resource_icon = branch.resource_icon == undefined ? branch.iconNameOverride : branch.resource_icon;
+
                           //if (!branch.children || branch.children.length === 0) {
                           //    tree_icon = attrs.iconLeaf;
                           //} else {
                           if (branch.expanded) {
                               tree_icon = attrs.iconCollapse;
                           } else if (branch.is_leaf) {
-                              tree_icon = attrs.iconLeaf;
+                              if (resource_icon == null) {
+                                  tree_icon = attrs.iconLeaf;
+                              } else {
+                                  // rather than showing the basic leaf icon _and_ the resource icon, just show the resource icon
+                                  tree_icon = resource_icon;
+                                  resource_icon = null;
```

Comment:
Hi @ahmelsayed 

Are you referring to the resource_icon = null line?

The standard icons for nodes are +/- for non-leaf nodes (depending on expanded status), and the document style icon for the leaf nodes.
![image](https://cloud.githubusercontent.com/assets/1824461/13155226/ac251f46-d674-11e5-91d9-a45e676aebec.png)

For non-leaf nodes I wanted the icon override to be in addition to the +/- icon for the treenode.
In the case of leaf nodes I wanted the icon override to replace the standard icon if specified rather than having it in addition to the document icon:

![image](https://cloud.githubusercontent.com/assets/1824461/13155331/17a47460-d675-11e5-8238-ab553b3de8a7.png)